lua/ttt2/libraries/outline.lua

```diff
@@ -20,37 +25,172 @@ OUTLINE_MODE_BOTH = 0
+    ["$smooth"] = 0
 }
+local outline_mat = CreateMaterial("ttt2_outline", "UnlitGeneric", outline_mat_settings)
 
-local copyMat = Material("pp/copy")
-local outlineMat = CreateMaterial("OutlineMat", "UnlitGeneric", outlineMatSettings)
-local storeTexture = render.GetScreenEffectTexture(0)
-local drawTexture = render.GetScreenEffectTexture(1)
+local copy_mat = Material("pp/copy")
 
 local ENTS = 1
 local COLOR = 2
 local MODE = 3
 
+--FIXME: You can decide if you want this to be a option for users, should be adressed before merge
+local enable_thin_line_workaround = false
+
+local function SetRenderType(render_type)
+    if (
+        render_type ~= OUTLINE_RENDERTYPE_BEFORE_VM and
+        render_type ~= OUTLINE_RENDERTYPE_BEFORE_EF and
+        render_type ~= OUTLINE_RENDERTYPE_AFTER_EF
+    ) then
+        return
+    end
+
+    RenderType = render_type
+
+    return true
+end
+
+local function GetRenderType()
+    return RenderType
+end
+
+
+local function GetOutlineThickness()
+    return OutlineThickness
+end
+
+local function SetOutlineThickness(thickness)
+    if thickness < 1 then
+        return
+    end
+
+    OutlineThickness = thickness
+
+    return true
+end
+
+
+local function GetCurrentLists()
+    return Lists[GetRenderType()]
+end
+
+local function GetCurrentListsSize()
+    return ListsSize[GetRenderType()]
+end
+
+
+local function GetCurrentList()
+    return Lists[GetRenderType()][GetOutlineThickness()]
+end
+
+local function GetCurrentListSize()
+    return ListsSize[GetRenderType()][GetOutlineThickness()]
+end
+
+
+local function ResetCurrentLists()
+    Lists[GetRenderType()] = {}
+end
+
+local function ResetCurrentListsSize()
+    ListsSize[GetRenderType()] = {}
+end
+
+
+local function SetCurrentList(value)
+    Lists[GetRenderType()] = value
+end
+
+local function SetCurrentListSize(size)
+    ListsSize[GetRenderType()][GetOutlineThickness()] = size
+end
+
+
+local function InitializeCurrentListIfNeeded()
+    local render_type, outline_thickness = GetRenderType(), GetOutlineThickness()
+
+    if not Lists[render_type] then
+        Lists[render_type] = {}
+    end
+
+    if not ListsSize[render_type] then
+        ListsSize[render_type] = {}
+    end
+
+    if not Lists[render_type][outline_thickness] then
+        Lists[render_type][outline_thickness] = {}
+    end
+
+    if not ListsSize[render_type][outline_thickness] then
+        ListsSize[render_type][outline_thickness] = 0
+    end
+end
+
 ---
 -- @param table ents List of @{Entity}
 -- @param Color color
 -- @param number mode [OUTLINE_MODE_BOTH, OUTLINE_MODE_NOTVISIBLE, OUTLINE_MODE_VISIBLE]
 -- @realm client
-function outline.Add(ents, color, mode)
+function outline.Add(ents, color, mode, render_type, outline_thickness)
+    -- Make the default behaviour comply with older revisions
+    if not render_type then
+        render_type = OUTLINE_RENDERTYPE_AFTER_EF
+    end
+
+    if not outline_thickness then
+        outline_thickness = 1
+    end
```

Comment:
is there a reason you didn't do `outline_thickness = outline_thickness or 1`, same for the other checks above

Reply:
fair, I don't care that much. I just assumed it had a performance impact and was chosen for that reason

Reply:
![Screenshot_20240922_232117](https://github.com/user-attachments/assets/5a1ec479-a98f-4f9f-986b-a3cbcff0b14e)
![Screenshot_20240922_232107](https://github.com/user-attachments/assets/99909660-e209-49b5-ae72-80d2cbd8e314)
Bytecode seems to be the exact same so either will do fine, just a style preference at that point

Reply:
Keep it as is, that's fine :)